.github/workflows/publish.yml

```diff
@@ -18,14 +18,27 @@ jobs:
     name: Publish CLI
     runs-on: ubuntu-latest
     steps:
+      - name: Wait for tests to succeed
+        uses: lewagon/wait-on-check-action@v1.0.0
```

Comment:
I'd still prefer to have this all in one workflow. Did you try using a conditional on the job to check the branch name?

```yaml
on: push

jobs:
  lint:
    ...
  test: 
    ...
  publish_cli:
    if: github.ref_name == 'main' || github.ref_name == 'rc' || startsWith(github.ref_name, 'arc-')
    needs: [lint, test]
```

Reply:
I did and that would work, I just really like the branch requirements up to so was trying to hold on to that. Will change it to this.